crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -133,16 +133,54 @@ impl Database for MongoDb {
             .expect("Failed to fetch latest job by given job type"))
     }
 
-    async fn get_successful_snos_jobs_without_proving(&self) -> Result<Vec<JobItem>> {
-        let filter = vec![
-            // Stage 1: Match successful SNOS job runs
+    /// function to get jobs that don't have a successor job.
+    ///
+    /// `job_a_type` : Type of job that we need to get that doesn't have any successor.
+    ///
+    /// `job_a_status` : Status of job A.
+    ///
+    /// `job_b_type` : Type of job that we need to have as a successor for Job A.
+    ///
+    /// `job_b_status` : Status of job B which we want to check with.
+    ///
+    /// Eg :
+    ///
+    /// Getting SNOS jobs that do not have a successive proving job initiated yet.
+    ///
+    /// job_a_type : SnosRun
+    ///
+    /// job_a_status : Completed
+    ///
+    /// job_b_type : ProofCreation
+    ///
+    /// job_b_status : Status of Job B / None
+    ///
+    /// **IMP** : For now Job B status implementation is pending so we can pass None
```

Comment:
let's not take job as a param for now because it allows others to call this function by mistake. and we can make the **IMP** a TODO then